addons/knowledge/static/src/js/wysiwyg.js

```diff
@@ -168,19 +169,20 @@ Wysiwyg.include({
         this._notifyNewBehaviors(container);
     },
     /**
-     * @private
+     * Insert a /article block (through a dialog)
      */
     _insertArticleLink: function () {
         const restoreSelection = preserveCursor(this.odooEditor.document);
         const dialog = new KnowledgeArticleLinkModal(this, {});
         dialog.on('save', this, data => {
             restoreSelection();
-            const link = QWeb.render('knowledge.wysiwyg_article_link', {
+            const linkHtml = $(QWeb.render('knowledge.wysiwyg_article_link', {
                 icon: data.icon,
                 text: data.text,
                 href: '/article/' + data.id
-            });
-            this.odooEditor.execCommand('insertHTML', link);
+            }))[0].outerHTML;
```

Comment:
I think that `QWeb.render` returns a string. So, I think you don't have to serialize the string to a JQuery object back to a string. Unless there is a catch ?

Reply:
the string returned by QWeb.render is sullied by \n text nodes which happen to be cleaned using the conversion in an element then taking the outerHtml value